### PR TITLE
Add logging for sql queries

### PR DESCRIPTION
**Pull request by jamestjw**
Don't do this at home folks, in a more serious project I would've never made all these changes without unit tests 😂

### CODE REVIEW

**Comment by jamestjw** [comment]
image

File: modules/models/record.php [hunk 1]
```diff
@@ -62,29 +63,34 @@ protected static function loadRecordFromData($data)
...
+            $sql = sprintf(
+                "SELECT * FROM %s WHERE %s",
+                $table_name,
+                implode(" AND ", $sql_wheres)
+            );
+        } else {
+            $sql = sprintf(
+                "SELECT * FROM %s;",
+                $table_name
+            );
```
Comment (jamestjw) [inline modules/models/record.php:90]: I hate the fact that I had to repeat this, if you guys have any ideas im all ears

File: modules/models/record.php [hunk 2]
```diff
@@ -93,23 +99,28 @@ public static function where($attrs)
         Usage: RecordName::find_by(array("name"=>"James", "student_id"=>12345));
         Returns the first record that matches
     */
-    public static function find_by($attrs)
+    public static function find_by(array $attrs)
```
Comment (jamestjw) [inline modules/models/record.php:102]: type annotation is always good

File: modules/models/record.php [hunk 3]
```diff
@@ -138,20 +149,14 @@ public function delete($pk="id") // Deletes by assuming $id is always the pk. No
...
-        }
-
         $sql = sprintf(
-            "DELETE FROM %s WHERE %s",
+            "DELETE FROM %s WHERE %s = :%s",
             get_called_class()::$table_name,
-            $condition
+            $pk,
+            $pk
         );
```
Comment (jamestjw) [inline modules/models/record.php:157]: if im not mistaken, using the colon allows us to take advantage of PDO's sql sanitisation (I havent verified this, its just a guess)

File: modules/models/record.php [hunk 4]
```diff
@@ -249,20 +255,21 @@ private function update($pk='id')
...
-                $value = sprintf("'%s'", $value);
-            }
-            array_push($update_fields, sprintf("%s = %s", $key, $value));
+        foreach (array_keys($to_update) as $key) {
+            array_push($update_fields, "$key = :$key");
         }
+
+        if (in_array("updated_at", get_called_class()::getAttrs())) {
+            array_push($update_fields, "updated_at = now()");
+        }
```
Comment (jamestjw) [inline modules/models/record.php:264]: Chris you missed this 😂

**Review by drouba — APPROVED** [review]
This seems all good to me

**Pull request merged**